imap_processing/hi/hi_l2.py

```diff
@@ -43,23 +47,34 @@ def hi_l2(
         Level 2 IMAP-Hi dataset ready to be written to a CDF file.
     """
     # TODO: parse descriptor to determine map configuration
-    sensor = "45" if "45" in descriptor else "90"
     direction: Literal["full"] = "full"
     cg_corrected = False
-    map_spacing = 4
+    map_descriptor = MapDescriptor.from_string(descriptor)
 
     rect_map = generate_hi_map(
         psets,
         geometric_factors_path,
         esa_energies_path,
         direction=direction,
+        output_map=map_descriptor.to_empty_map(),
         cg_corrected=cg_corrected,
-        map_spacing=map_spacing,
     )
 
     # Get the map dataset with variables/coordinates in the correct shape
     # TODO get the correct descriptor and frame
-    l2_ds = rect_map.build_cdf_dataset("hi", "l2", "sf", descriptor, sensor=sensor)
+
+    if not isinstance(rect_map, RectangularSkyMap):
+        raise NotImplementedError("HEALPix map output not supported for Hi")
+    if not isinstance(map_descriptor.sensor, str):
+        raise NotImplementedError("Sensor must be 45 or 90")
```

Comment:
Please change this to a `ValueError` and specify that it was not found to be a string.
```suggestion
        raise ValueError("Invalid map_descriptor. Sensor attribute must be of type str and be either '45' or '90'")
```